### PR TITLE
ppc64 syscall, perf tests with not 4k pages

### PR DESCRIPTION
**Pull request by arthurfabre**
Add the ppc64 BPF syscall definition.

Fix the perf lost sample test to not rely on 4K page sizes.

### CODE REVIEW

File: perf_test.go [hunk 1]
```diff
@@ -89,6 +181,26 @@ func TestPerfReaderLostSample(t *testing.T) {
 	}
 }
 
+// TODO - xdpcap.Hook.Patch() uses the same thing, should we add a CollectionSpec.UseMap?
+// Or maybe just ProgramSpec.RewriteMap(), and have NewCollection() not rewrite already rewritten maps.
```
Comment (arthurfabre) [inline perf_test.go]: We've talked about this before, but I can't remember the conclusion.

File: perf_test.go [hunk 2]
```diff
@@ -46,28 +47,118 @@ func TestPerfReader(t *testing.T) {
...
+	//
+	//  - PERF_RECORD_LOST: 8 (perf_event_header) + 16 (PERF_RECORD_LOST)
+	//
+	//  - output_single: 8 (perf_event_header) + 4 (size) + 5 (payload) + 7 (padding to 64bits)
+	//
+	// By selecting an event size of the form 2^n + 2^(n+1), for any page size 2^(n+m), m >= 0,
+	// the number of bytes left, x, after filling a page with a whole number of events is:
+	//
+	//                     2^(n+m)                            2^n * 2^m
+	//  x = 2^n * frac(---------------) <=> x = 2^n * frac(---------------)
```
Comment (lmb) [inline perf_test.go]: Okeeey dokey :+1:

File: testdata/perf_output.c [hunk 3]
```diff
@@ -14,26 +14,9 @@ __section("xdp/single") int output_single(void *ctx) {
...
-		if (ret) {
-			return ret;
-		}
-	}
-
-	// Lost sample events are generated opportunistically, when the kernel
-	// is writing an event and realises that there were events lost previously.
-	// Generate a small event to trigger this.
-	return perf_event_output(ctx, &events, cpu, &buf[0], 1);
+	return perf_event_output(ctx, &events, BPF_F_CURRENT_CPU, &buf[0], 180); // TODO - use sizeof()
```
Comment (lmb) [inline testdata/perf_output.c]: TODO: TODO

File: perf_test.go [hunk 4]
```diff
@@ -46,28 +47,118 @@ func TestPerfReader(t *testing.T) {
...
+		t.Fatal("unsupported page size:", pageSize)
+	}
+
+	spec, err := LoadCollectionSpec("testdata/perf_output.elf")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Single perf map for output_large and output_single
+	perfMap, err := NewMap(&MapSpec{
```
Comment (lmb) [inline perf_test.go]: Couldn't you just DetachMap after NewCollection?

File: perf_test.go (hunk 1 above)
Comment (lmb) [reply perf_test.go]: Yeah, I think it makes sense for RewriteMap to return a new error if insn.Constant != 0, and then catch that in NewCollection. See my comment above though, maybe we don't need this?

File: syscalls_ppc64le.go [hunk 5]
```diff
@@ -0,0 +1,3 @@
+package ebpf
+
+const _BPFCall = 361
```
Comment (lmb) [inline syscalls_ppc64le.go]: Next task: add a builder for this ;P

File: testdata/perf_output.c (hunk 3 above)
Comment (arthurfabre) [reply testdata/perf_output.c]: Oops yes, fixed

File: perf_test.go (hunk 4 above)
Comment (arthurfabre) [reply perf_test.go]: Duh, I completely zapped that one ELF, with two entrypoints == one map.. Fixed!

File: syscalls_ppc64le.go (hunk 5 above)
Comment (arthurfabre) [reply syscalls_ppc64le.go]: Haha at least we could test the online / available CPU parsing :P

```
arthur@steve ~/g/s/g/n/ebpf> cat /sys/devices/system/cpu/possible
0-143
arthur@steve ~/g/s/g/n/ebpf> cat /sys/devices/system/cpu/online
0-143
arthur@steve ~/g/s/g/n/ebpf> sudo ppc64_cpu --cores-on=1
arthur@steve ~/g/s/g/n/ebpf> cat /sys/devices/system/cpu/online
0-3
arthur@steve ~/g/s/g/n/ebpf> sudo ppc64_cpu --online-cores=6,8
arthur@steve ~/g/s/g/n/ebpf> cat /sys/devices/system/cpu/online
0-3,24-27,32-35
arthur@steve ~/g/s/g/n/ebpf> sudo ppc64_cpu --cores-on=all
arthur@steve ~/g/s/g/n/ebpf> cat /sys/devices/system/cpu/online
0-143
```

File: perf_test.go (hunk 1 above)
Comment (arthurfabre) [reply perf_test.go]: Yup, turns out we don't need this at all. Sounds sane enough :)

**Pull request merged**